src/AutoRest.CSharp/Generation/Writers/ClientWriter.cs

```diff
@@ -18,6 +22,65 @@ namespace AutoRest.CSharp.Generation.Writers
 {
     internal class ClientWriter
     {
+
+        public static void WriteClientOptions(CodeWriter writer, Client client, BuildContext context)
```

Comment:
Let's have a separate writer per type we produce and move this into ClientOptionsWriter.